logfmt_test.go

```diff
@@ -17,56 +18,74 @@ const (
 func TestAppendLogfmt(t *testing.T) {
 	t.Parallel()
 
-	b := make([]byte, 0, 4096)
-	b, _ = appendLogfmt(b, nil)
+	buf := make([]byte, 0, 4096)
+
+	b, _ := appendLogfmt(buf, nil)
 	if string(b) != "null" {
 		t.Error(string(b) + " != null")
 	}
 
-	b = b[:0]
-	b, _ = appendLogfmt(b, 100)
+	b, _ = appendLogfmt(buf, 100)
 	if string(b) != "100" {
 		t.Error(string(b) + " != 100")
 	}
 
-	b = b[:0]
-	b, _ = appendLogfmt(b, false)
+	b, _ = appendLogfmt(buf, false)
 	if string(b) != "false" {
 		t.Error(string(b) + " != false")
 	}
 
-	b = b[:0]
-	b, _ = appendLogfmt(b, true)
+	b, _ = appendLogfmt(buf, true)
 	if string(b) != "true" {
 		t.Error(string(b) + " != true")
 	}
 
-	b = b[:0]
-	b, _ = appendLogfmt(b, `"abc `)
+	b, _ = appendLogfmt(buf, `"abc `)
 	if string(b) != `"\"abc "` {
 		t.Error(string(b) + ` != "\"abc "`)
 	}
 
-	b = b[:0]
-	b, _ = appendLogfmt(b, []int{-100, 100, 20000})
+	b, _ = appendLogfmt(buf, float32(3.14159))
+	if string(b) != "3.14159" {
+		t.Error(`string(b) != "3.14159"`)
```

Comment:
It would be more friendly to show the erroneous result, though not all the tests do so.

```suggestion
		t.Error(string(b) + ` != "3.14159"`)
```